src/modules/app.ts

```diff
@@ -339,18 +339,21 @@ export class AppModule
    * @returns - The sanitized metadata with an uploaded image ipfs hash
    */
   private async _prepareMetadata(metadata: CommonModuleMetadata): Promise<any> {
-    if (typeof metadata.image === "string") {
-      return Promise.resolve(metadata);
-    }
-    if (metadata.image === undefined) {
-      return Promise.resolve(metadata);
+    for(const key in metadata) {
```

Comment:
Break out this logic into its own function so it can be called recursively. Name it `_handleFileProperties` (private)